### PR TITLE
[App] Wrap LightningClient methods with retry wrapper by default

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
Addresses https://github.com/gridai/grid/pull/9033#discussion_r1071372526.

<details><summary>This PR can impact these occurrences</summary>

```console
$ git grep "LightningClient(" src/lightning_app | grep -v "import "
src/lightning_app/cli/cmd_apps.py:        self.api_client = LightningClient()
src/lightning_app/cli/cmd_clusters.py:        self.api_client = LightningClient()
src/lightning_app/cli/cmd_ssh_keys.py:        self.api_client = LightningClient()
src/lightning_app/cli/commands/connection.py:                client = LightningClient()
src/lightning_app/cli/commands/logs.py:    client = LightningClient()
src/lightning_app/cli/lightning_cli.py:    client = LightningClient()
src/lightning_app/runners/backends/cloud.py:        self.client = LightningClient(retry=True)
src/lightning_app/testing/testing.py:    client = LightningClient()
src/lightning_app/testing/testing.py:        client = LightningClient()
src/lightning_app/testing/testing.py:    client = LightningClient()
src/lightning_app/utilities/app_helpers.py:    client = LightningClient()
src/lightning_app/utilities/cli_helpers.py:        client = LightningClient()
src/lightning_app/utilities/commands/base.py:                client = LightningClient()
src/lightning_app/utilities/network.py:class LightningClient(GridRestClient):
src/lightning_app/utilities/port.py:    client = LightningClient()
src/lightning_app/utilities/port.py:    client = LightningClient()
src/lightning_app/utilities/port.py:    client = LightningClient()
src/lightning_app/utilities/secrets.py:    lightning_client = LightningClient()
```

</details>

### Does your PR introduce any breaking changes? If yes, please list them.
Changes `retry=False` to `retry=True`:
```diff
- lightning_app.utilities.network.LightningClient(retry=False)
+ lightning_app.utilities.network.LightningClient(retry=True)
```

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @tchaton

### CODE REVIEW

**Comment by awaelchli** [comment]
@akihironitta Please have a look at the motivation from #15412. There are some consequences in the CLI to doing so.

**Review by Borda — APPROVED** [review]

**Comment by akihironitta** [comment]
@tchaton What would you suggest here? Should we still have `retry=True` as the default value and turn it off for CLI-related places to avoid the issue described in #15412?

**Review by tchaton — APPROVED** [review]
LGTM !

**Comment by akihironitta** [comment]
Given https://github.com/Lightning-AI/lightning/pull/16382#issuecomment-1384290226, I see two options:
1. Close this PR
2. Disable `retry` in CLI-related codebase and merge this PR

I'll mark this PR as ready for review again once I update this PR (2.).

**Comment by awaelchli** [comment]
@akihironitta 2) sounds good to me. To be sure, I would also check the usages in other places than the CLI and see if the default value make sense or whether it should be flipped too.

**Review by awaelchli — APPROVED** [review]

File: tests/tests_app/utilities/test_network.py [hunk 1]
```diff
@@ -9,9 +9,9 @@ def test_port():
 
 def test_lightning_client_retry_enabled():
     with patch("lightning_app.utilities.network._retry_wrapper") as wrapper:
-        LightningClient()  # default: retry=False
+        LightningClient(retry=False)
         wrapper.assert_not_called()
```
[the remark this answers is not in the document]
Comment (tchaton) [reply tests/tests_app/utilities/test_network.py]: Done ! Removed the _new__ as setattr on cls leads to retry=False to be already wrapped.

**Review by awaelchli — APPROVED** [review]

**Pull request merged**